### PR TITLE
Fix NXP 21336 fix cell separator

### PR DESCRIPTION
**Pull request by richardsd**
Reopened the PR due to some issues with rebase.
Added a unit test for the cell separator.
Will backport it to 8.10 too.

### CODE REVIEW

File: nuxeo-core/nuxeo-core-convert-plugins-test/src/test/java/org/nuxeo/ecm/core/convert/plugins/tests/advanced/TestAdvancedXLX2TextConversion.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    @Test
+    public void testCellSeparator() throws Exception {
+        doTestTextConverter("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", "xlx2text",
+                "advanced/cell_separator.xlsx");
+    }
+
+    @Override
+    protected void checkTextConversion(String textContent) {
+        assertTrue(!textContent.isEmpty());
+        String[] cells = textContent.trim().split(separator);
```
Comment (efge) [inline nuxeo-core/nuxeo-core-convert-plugins-test/src/test/java/org/nuxeo/ecm/core/convert/plugins/tests/advanced/TestAdvancedXLX2TextConversion.java]: Just use `" "` here, no need for a constant.

File: nuxeo-core/nuxeo-core-convert-plugins-test/src/test/java/org/nuxeo/ecm/core/convert/plugins/tests/advanced/TestAdvancedXLX2TextConversion.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        assertTrue(cells.length == 2);
+        assertArrayEquals(cells, new String[] { "hello", "world" });
```
Comment (efge) [inline nuxeo-core/nuxeo-core-convert-plugins-test/src/test/java/org/nuxeo/ecm/core/convert/plugins/tests/advanced/TestAdvancedXLX2TextConversion.java]: The first `assertTrue` is not needed, it's redundant with the full array equality test.

**Comment by efge** [comment]
👍

**Comment by nelsonsilva** [comment]
:+1:

**Pull request closed without merging**